elastic/kibana
# [Event Log][7.16] Set all event log index assets to hidden

resolves https://github.com/elastic/kibana/issues/116341

## Summary

For 8.0.0, PR https://github.com/elastic/kibana/pull/110929 changed the event log to be a hidden index.  However, in 7.16.0, an elasticsearch deprecation is logged when indices are created whose name starts with `.` (like the event log) are not hidden.  To avoid the deprecation, this PR is a cherry-pick of PR #110929.

### To Verify

See the [**To Verify** section of PR 110929](https://github.com/elastic/kibana/pull/110929).

In addition, check data stream `.logs-deprecation.elasticsearch-default` to ensure the following message is not logged as a document, after Kibana finishes startup:

`index name [.kibana-event-log-7.16.0-000001] starts with a dot '.', in the next major version, index names starting with a dot are reserved for hidden indices and system indices`

### Checklist


## Release note

The Alerting event log indices, named `kibana-event-log-*`, are set to hidden indices via the `index.hidden: true` index setting.
